File: cpp/Chapter I/Strings/escape.cpp
// ACCESS STRING IN C++
// Note: String indexes start with 0: 
// [0] is the first character. [1] is the second character, etc.


#include <iostream>
using namespace std;

int main() {
    // Declare and initialize a string variable 
    // named 'message' with the value "Hello, World"

    string message = "Hello, World"; 
    cout << message << endl;

    // You can access the characters in a string by 
    // referring to its index number inside square brackets [].
    string greeting = "How are You?";
    cout << greeting[2] << endl; // output w

    //CHANGING STRING VALUES
    greeting[11] = '!'; // change ? to !
    cout << greeting;  // Output How are You! 

    // ESCAPE CHARACTER IN C++:
    
    // In C++, escape characters are special sequences of characters that 
    // represent non-printable or special characters when used within strings. 
    // They are preceded by a backslash \ to distinguish them from regular characters. 
    // Here are some commonly used escape characters in C++:

   // \n: Newline character. It moves the cursor to the beginning of the next line.
   // \t: Tab character. It inserts a horizontal tab.
   //  \": Double quote character. It allows you to include double quotes within a string.
   //  \': Single quote character. It allows you to include single quotes within a string.
   //  \\: Backslash character. It allows you to include backslashes within a string.

    cout << "This is a new line.\n"; // Newline character
    cout << "This is a tab\tcharacter.\n"; // Tab character
    cout << "This is a double quote: \"Hello\"\n"; // Double quote character
    cout << "This is a single quote: \'a\'\n"; // Single quote character
    cout << "This is a backslash: \\ \n"; // Backslash character
    return 0;
}

File: python/chapter_02/chapter_02b/Strings/Escape.py
# ESCAPE CHARACTERS IN PYTHON

# Escape characters in Python are special characters that are preceded by a backslash (\).
# They are used to represent characters that are difficult or impossible to type directly into a string.
# Here are some common escape characters and their explanations:
#
#     \n: Newline - Inserts a newline character.
#     \t: Tab - Inserts a tab character.
#     \\: Backslash - Inserts a backslash character.
#     \": Double quote - Inserts a double quote character.
#     \': Single quote - Inserts a single quote character.

# Example demonstrating escape characters
# Newline character
print("Hello\nworld!")  # Output: Hello
                         #         world!

# Tab character
print("Hello\tworld!")  # Output: Hello   world!

# Backslash character
print("This is a backslash: \\")  # Output: This is a backslash: \

# Double quote character
print("He said, \"Hello!\"")  # Output: He said, "Hello!"


# Single quote character
print('It\'s raining')  # Output: It's raining
message = 'One of Python\'s strengths is its diverse community.'
print(message)

# COMBINED ESCAPE CHARACTERS
print("Languages:\n\tPython\n\tC\n\tJavaScript")